src/libcollections/string.rs

```diff
@@ -138,6 +138,7 @@ impl String {
     /// let output = String::from_utf8_lossy(input);
     /// assert_eq!(output.as_slice(), "Hello \u{FFFD}World");
     /// ```
+    #[inline]
```

Comment:
If there were an attribute #[can_be_inlined] I would have used that. Adding #[inline] is the only way to let llvm decide if it should be inlined.